doc/04-Upgrading.md

```diff
@@ -3,6 +3,64 @@
+time to perform the reindexing on the `history` table even if it is not blocked. Nonetheless, you can use the
+`show processlist` command to determine whether an excessive number of queries have been stuck in a waiting state.
+
+```
+MariaDB [icingadb]> show processlist;
++------+-----+-----+----------+-----+------+---------------------------------+------------------------------------+-----+
+| Id   | ... | ... | db       | ... | Time | State                           | Info                               | ... |
++------+-----+-----+----------+-----+------+---------------------------------+------------------------------------+-----+
+| 1475 | ... | ... | icingadb | ... | 1222 | Waiting for table metadata lock | INSERT INTO "notification_history" | ... |
+| 1485 | ... | ... | icingadb | ... | 1262 | Creating sort index             | SELECT history.id, history....     | ... |
+| 1494 | ... | ... | icingadb | ... | 1224 | Waiting for table metadata lock | ALTER TABLE history ADD INDEX ...  | ... |
+| 1499 | ... | ... | icingadb | ... | 1215 | Waiting for table metadata lock | INSERT INTO "notification_history" | ... |
+| 1500 | ... | ... | icingadb | ... | 1215 | Waiting for table metadata lock | INSERT INTO "state_history" ...    | ... |
++------+-----+-----+----------+-----+------+---------------------------------+------------------------------------+-----+
+39 rows in set (0.001 sec)
+```
+
+In the above output you can see that there are approximately `39` queries (`39 rows in set (0.001 sec)`), including the
+`ALTER TABLE history ADD INDEX` query from the upgrade script, waiting for a metadata lock, they're just minimised to
+the bare essentials. Unfortunately, only one of these queries is holding the `table metadata lock` that everyone else
+is now waiting for, which in this case is a `SELECT` statement initiated by Icinga DB Web in the `icingadb/history`
+view, which takes an unimaginably long time. Note that there might be multiple `SELECT` statements started before the
+upgrade script in your case when the history view of your Icinga DB Web is opened by different Icinga Web users at the
+same time.
+
+You can now either just wait for the `SELECT` statements to finish by themselves and let them block the upgrade script
+and all Icinga DB queries on all `*_history` tables or forcibly terminate them and let the remaining queries do their
+work. In this case, cancelling that one blocking `SELECT` query will let the upgrade script continue normally without
+blocking any other queries.
+``` 
+MariaDB [icingadb]> kill 1485;
+```
+
+In case you are insecure about which Icinga DB Web queries are blocking, you may simply cancel all long-running
+`SELECT` statements listed with `show processlist` (see column `Time`). Cancelling a `SELECT` query is usually not that
+disastrous, so feel free to abort every single one of them matching the Icinga DB database (see column `db`).
```

Comment:
> usually not that disastrous

That wording isn't really reassuring if I'm already unsure. I mean what's the worst that could happen? The same as if the database was stopped at that moment.

Reply:
> I mean what's the worst that could happen? The same as if the database was stopped at that moment.

Is that really that important? We already retry all database errors anyway. So, how badly could Icinga DB be damaged if it had to repeat its queries once again?

Reply:
My comment is about the "usually not that disastrous" wording specifically. Isn't the whole point of this sentence to reassure users that killing a `SELECT` query is fine? But with this wording, one may ask oneself why it says "usually" and how one can be sure that this isn't an unusual situation where it is disastrous.

> I mean what's the worst that could happen? The same as if the database was stopped at that moment.

That's not what I would write into the docs. That's just my justification why I don't think we need to use very careful wording here. Killing a query is no worse than stopping the database, or even just losing the network connection. If that would be disastrous, you have other problems.